src/hotspot/cpu/x86/c2_MacroAssembler_x86.cpp

```diff
@@ -812,7 +812,7 @@ void C2_MacroAssembler::fast_unlock(Register objReg, Register boxReg, Register t
     jccb(Assembler::zero, Stacked);
     if (UseFastLocking) {
       // If the owner is ANONYMOUS, we need to fix it - in the slow-path.
-      testptr(Address(tmpReg, OM_OFFSET_NO_MONITOR_VALUE_TAG(owner)), (int32_t) (intptr_t) ANONYMOUS_OWNER);
+      testb(Address(tmpReg, OM_OFFSET_NO_MONITOR_VALUE_TAG(owner)), (int32_t) (intptr_t) ANONYMOUS_OWNER);
```

Comment:
Sanity question: This would accept the monitor that has the lowest byte equal to `ANONYMOUS_OWNER`. Are we guaranteed that no monitor would ever have the lowest bit as `1`? Probably true, just checking the reason here.

Reply:
Valid monitor owners can only be Thread* or ANONYMOUS or NULL (or stack-pointers, but only with stack-locking). Thread instances are allocated at least word-aligned, so no, 1 cannot be a valid monitor owner.